community/consistency-check/src/main/java/org/neo4j/consistency/CheckConsistencyCommand.java

```diff
@@ -0,0 +1,160 @@
+import org.neo4j.io.fs.DefaultFileSystemAbstraction;
+import org.neo4j.kernel.configuration.Config;
+import org.neo4j.kernel.impl.util.Converters;
+import org.neo4j.logging.FormattedLogProvider;
+import org.neo4j.server.configuration.ConfigLoader;
+
+import static org.neo4j.dbms.DatabaseManagementSystemSettings.database_path;
+
+public class CheckConsistencyCommand implements AdminCommand
+{
+    public static class Provider extends AdminCommand.Provider
+    {
+        public Provider()
+        {
+            super( "check-consistency" );
+        }
+
+        @Override
+        public Optional<String> arguments()
+        {
+            return Optional.of( "--database=<database> [--additional-config=<file>] [--verbose]" );
+        }
+
+        @Override
+        public String description()
+        {
+            return "Check the consistency of a database.";
+        }
+
+        @Override
+        public AdminCommand create( Path homeDir, Path configDir, OutsideWorld outsideWorld )
+        {
+            return new CheckConsistencyCommand( homeDir, configDir );
+        }
+    }
+
+    private final Path homeDir;
+    private final Path configDir;
+    private final ConsistencyCheckService consistencyCheckService;
+
+    public CheckConsistencyCommand( Path homeDir, Path configDir )
+    {
+        this( homeDir, configDir, new ConsistencyCheckService() );
+    }
+
+    public CheckConsistencyCommand( Path homeDir, Path configDir, ConsistencyCheckService consistencyCheckService )
+    {
+        this.homeDir = homeDir;
+        this.configDir = configDir;
+        this.consistencyCheckService = consistencyCheckService;
+    }
+
+    @Override
+    public void execute( String[] args ) throws IncorrectUsage, CommandFailed
+    {
+        String database;
+        Boolean verbose;
+        File additionalConfigFile;
+
+        Args parsedArgs = Args.parse( args );
+        try
+        {
+            database = parsedArgs.interpretOption( "database", Converters.mandatory(), s -> s );
+            verbose = parsedArgs.getBoolean( "verbose" );
+            additionalConfigFile =
+                    parsedArgs.interpretOption( "additional-config", Converters.optional(), Converters.toFile() );
+        }
+        catch ( IllegalArgumentException e )
+        {
+            throw new IncorrectUsage( e.getMessage() );
+        }
+
+        Config config = loadNeo4jConfig( homeDir, configDir, database, loadAdditionalConfig( additionalConfigFile ) );
+
+        try
+        {
+            consistencyCheckService.runFullConsistencyCheck( config.get( database_path ), config,
+                    ProgressMonitorFactory.textual( System.err ), FormattedLogProvider.toOutputStream( System.out ),
+                    new DefaultFileSystemAbstraction(), verbose );
+        }
+        catch ( ConsistencyCheckIncompleteException | IOException e )
+        {
+            throw new CommandFailed( "Consistency checking failed." + e.getMessage(), e );
```

Comment:
I think you need a space after the full stop in there, or it will be hard to read.